src/ongoing_takers.rs

```diff
@@ -0,0 +1,78 @@
+use crate::network::Taker;
+use std::collections::HashSet;
+
+// TODO: Find a better name
+#[derive(Default, Debug)]
+pub struct OngoingTakers(HashSet<Taker>);
+
+impl OngoingTakers {
+    pub fn insert(&mut self, taker: Taker) -> anyhow::Result<()> {
+        if self.0.contains(&taker) {
+            anyhow::bail!("taker {} is already part of ongoing takers", taker.0)
+        }
+
+        self.0.insert(taker);
+        Ok(())
+    }
+
+    pub fn cannot_trade_with_taker(&self, taker: &Taker) -> bool {
```

Comment:
[Rust's naming convention](https://doc.rust-lang.org/1.0.0/style/style/naming/README.html) is that boolean predicates should be prefixed with `is_` or other question word. It's a good practice because the current name is confusing.
```suggestion
    pub fn has_an_ongoing_trade(&self, taker: &Taker) -> bool {
```